djstripe/models.py

```diff
@@ -542,35 +542,29 @@ def _attach_objects_hook(self, cls, data):
         if subscription:
             self.subscription = subscription
 
-        self._attach_invoice_items(cls, data)
-
-    def _attach_invoice_items(self, cls, data):
-        if not self.pk:
-            # InvoiceItems need a saved invoice because they're associated via
-            # a RelatedManager.  It might be better to make this pattern more
-            # generic with some sort-of _attach_objects_post_save_hook().
-            self.save()
-
+    def _attach_objects_post_save_hook(self, cls, data):
+        # InvoiceItems need a saved invoice because they're associated via a
+        # RelatedManager, so this must be done as part of the post save hook.
         cls._stripe_object_to_invoice_items(InvoiceItem, data, self)
 
     @property
     def plan(self):
         """ Gets the associated plan for this invoice.
 
         In order to provide a consistent view of invoices, it is necessary to
-        examine the invoice items for the associated subscription plan rather
-        than the top-level link.  The reason for this is that the subscription
-        plan when requested by the customer, but the invoice item plan will
-        remain the same.  This makes it difficult to work with an invoice
-        history if the plan is expected to remain consistent (e.g. for creating
-        downloadable invoices).
+        extract it from the first invoice item with a plan, rather than getting
+        it from the subscription by foreign key.  The reason for this is that
+        the subscription plan when requested by the customer, but the invoice
```

Comment:
Sorry, meant second sentence of this paragraph, but the clarification you've made helps.

The reason for this is the subscription plan ...(what?)... when requested by the customer